pages/samvera/developer_resources/collections/hyrax_2.1/nested-indexing.md

```diff
@@ -0,0 +1,55 @@
+---
+title: "Nested Indexing"
+keywords: Indexing, Collection
+categories: How to Do All the Things
+permalink: nested-indexing.html
+folder: samvera/how-to/collections/hyrax_2.1/nested-indexing.md
+sidebar: home_sidebar
+tags: [development_resources]
+a-z: ['Nested Indexing', 'Collections - Indexing']
+version:
+  label: 'targeted for Hyrax v2.1.0'
+  branch:
+    label: 'master'
+    link: 'https://github.com/samvera/hyrax'
+---
+
+### Collections In Fedora and Solr
+In Hyrax, a child (work or collection) is stored as a member of a parent collection. Fedora includes an LDP container “member_of_collections” and a memberOf relationship.
+
+The collection relationship is indexed in Solr into "member_of_collection_ids_ssim" and “member_of_collections_ssim” as part of the standard to_solr method. Four additional fields (as defined in the configuration settings) are added to Solr via an after_update_index callback.
+
+### In the Model
+Nested Indexing is added to Hyrax via the inclusion of Hyrax::CollectionNesting in the Collection and Work models.
+
+### Nested Indexing Details
+Hyrax::CollectionNesting extends ActiveSupport::Concern to add an after_update_index callback. This calls the Samvera::NestingIndexer via Hyrax’s nesting_index_adapter.
+
+The nesting indexer updates the Solr document of the object with four additional fields which are used for displaying the nested collections and validating which collections are valid for new creating nested relationships. This indexing is based on the object's "member_of_collections" attribute stored in ActiveFedora.
+
+A call to the indexer reindexes the object itself, as well as all of its descendant collections and works, making this an expensive transaction. For this reason, the to_solr method does not load these fields.
+
```

Comment:
Do you mean to_solr will never include these fields, or just not until they're available?

Reply:
Yes, to_solr method NEVER includes the fields. To_solr methods begin with a Fedora object and convert each field into its solr equivalent. Most objects have their to_solr methods overridden to add additional data that isn't part of the Fedora saved object, but we don't want these fields loaded every time to_solr gets called. If the data is needed once it is loaded, a separate query is needed to retrieve the solr document.